### DISCUSSION TITLE
Mac version please?

### DISCUSSION THREAD

**Discussion by zumzum360** [ntsc-rs/ntsc-rs#131] [General]
Can you all make a mac version? because i try EVERY SINGLE TIME TO FIND A VHS APP and can you port it on mac... for M1/intel or make it universal? because i want to make VHS videos on it.... without using windows or Whisky though.... can you make it for mac?

**Comment by Degamisu** [comment]
#36 has been an ongoing issue and is being actively worked on. From what I've read, you would need to build it yourself. Here are the provided lines you should put in your terminal

```bash
rustup install stable
brew install --cask gstreamer-runtime
brew install --cask gstreamer-development
git clone --recurse-submodules https://github.com/valadaptive/ntsc-rs.git
cd ntsc-rs
git checkout macos
export PATH=$PATH:/Library/Frameworks/GStreamer.framework/Versions/1.0/bin
export PKG_CONFIG_PATH=/Library/Frameworks/GStreamer.framework/Versions/1.0/lib/pkgconfig
# To build the standalone app (output will be in the `build` folder)
cargo xtask macos-bundle --release

# Alternatively, you can build the OFX plugin:
cargo xtask build-ofx-plugin --macos-universal --release

# Or the AE plugin.. (untested)
cargo xtask macos-ae-plugin --macos-universal --release
```

>[!NOTE]
>If needed, you may need codesigning for the AE plugin

```bash
codesign --options runtime --timestamp -strict --sign - build/ntsc-rs.plugin
```

Let me know if this works!

**Reply by zumzum360** [reply]
.....could you kinda turn this into a app? because i don't feel like using brew

**Reply by Degamisu** [reply]
> .....could you kinda turn this into a app? because i don't feel like using brew

What do you mean by that? If you want me to upload the artifacts, I'll try my best to edit the Github Actions. I am not a Mac user, so I cannot compile it directly.

**Reply by valadaptive** [reply]
@Degamisu If you want to build it for macOS, you'll need to use the "macos" branch right now. Although I've posted builds below, so you don't have to.

**Comment by valadaptive** [comment]
I'm making good progress on the macOS version, and it'd be great if you could help test it.

[Here's the latest build](https://drive.proton.me/urls/DYFNPH9C1R#HxGgpqV8Cu0K). Let me know:

- Does it run when you double-click it (with a "this was downloaded from the internet" warning), or does it refuse to run until you manually allow it to in the Privacy & Security settings?
- Does it start up properly? Give it a few minutes to get past the loading screen.
- Can you load a variety of image and video formats without errors? If you do run into any errors, screenshot them and post them here.

**Reply by zumzum360** [reply]
Thanks

**Reply by zumzum360** [reply]
it did work

**Reply by damonkex** [reply]
Yo I super appreciate your efforts to make a macOS version possible, I'll be testing it a whole bunch soon

Specs:
* MacBook Pro M1
* macOS Sonoma 14.5

So far:
* It runs, but I have to Option + Click to actually open it
* It seems to start up properly and very quickly after Option + Click
* I haven't ran into any errors yet, but I've only tried importing .mov & .mp4
